traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -1055,52 +1065,56 @@ func Update(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	rows, err := inf.Tx.NamedQuery(updateQuery, server)
-	if err != nil {
-		userErr, sysErr, errCode = api.ParseDBError(err)
-		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
-		return
-	}
-	defer rows.Close()
+	if changeXMPPID {
+		api.WriteRespAlertObj(w, r, tc.ErrorLevel, fmt.Sprintf("server cannot be updated due to requested XMPPID change. XMPIDD is immutable"), nil)
```

Comment:
You should return here, after `api.WriteRespAlertObj` because of two reasons:
1.) That way, you dont have to use the `else` block
2.) If you dont return, the `error.log` will complain about response being written twice

Reply:
:) Simplified it.